packages/plugin/src/Integrations/PaymentGateways/PayPal/Controllers/OrdersController.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\Controllers;
+
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\Services\PayPalOrderService;
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\Services\PayPalPriceService;
+use Solspace\Freeform\Library\Helpers\IsolatedTwig;
+use yii\web\NotFoundHttpException;
+use yii\web\Response;
+
+class OrdersController extends BasePayPalController
+{
+    public $enableCsrfValidation = false;
+    protected array|bool|int $allowAnonymous = true;
+
+    public function __construct(
+        $id,
+        $module,
+        $config,
+        private IsolatedTwig $isolatedTwig,
+        private PayPalOrderService $orderService,
+        private PayPalPriceService $priceService,
+    ) {
+        parent::__construct($id, $module, $config);
+    }
+
+    public function actionCreate(): Response
+    {
+        try {
+            [$form, $integration, $field, $hash] = $this->getRequestItems(handleFormRequest: false);
+
+            // Validate integration
+            if (!$integration->isEnabled()) {
+                throw new \Exception('PayPal integration is not enabled');
+            }
+
+            if (!$integration->getClientId() || !$integration->getClientSecret()) {
+                throw new \Exception('PayPal integration credentials are not configured');
+            }
+        } catch (NotFoundHttpException $exception) {
+            return $this->asSerializedJson(['errors' => [$exception->getMessage()]], 404);
+        } catch (\Exception $exception) {
+            return $this->asSerializedJson(['errors' => [$exception->getMessage()]], 400);
+        }
+
+        $description = $this->isolatedTwig
+            ->render(
+                $field->getDescription(),
+                [
+                    'form' => $field->getForm(),
+                    'field' => $field,
+                ]
+            )
+        ;
+
+        // Allow dynamic amount: merge posted values (from client) into form before calculating
+        $raw = $this->request->getRawBody();
+        $json = $raw ? json_decode($raw, true) : null;
+        $values = \is_array($json) && isset($json['values']) && \is_array($json['values']) ? $json['values'] : null;
+        if ($values) {
+            foreach ($form->getLayout()->getFields()->getIterator() as $f) {
+                $handle = $f->getHandle();
+                if (\array_key_exists($handle, $values)) {
+                    $f->setValue($values[$handle]);
+                }
+            }
```

Comment:
Please refrain from using `$f`, `$a`, `$b` and whatever else shorthands anywhere. It may make sense to use for these quick loops, but it is never worth it in the long run.

Also - `::getFields()` returns a `FieldCollection` which is an iterable object and you don't need to access the iterator for iterating over it.

```suggestion
            foreach ($form->getLayout()->getFields() as $formField) {
                $handle = $formField->getHandle();
                if (\array_key_exists($handle, $values)) {
                    $formField->setValue($values[$handle]);
                }
            }
```